vscode-trace-extension/src/trace-explorer/properties/trace-explorer-properties-view-webview-provider.ts

```diff
@@ -33,26 +33,21 @@ export class TraceExplorerItemPropertiesProvider extends AbstractTraceExplorerPr
                 }
             }
         });
-        signalManager().on(Signals.ITEM_PROPERTIES_UPDATED, this.handleUpdatedProperties);
-        signalManager().on(Signals.EXPERIMENT_SELECTED, this.handleExperimentChanged);
-        signalManager().on(Signals.CLOSE_TRACEVIEWERTAB, this.handleTabClosed);
+        signalManager().on('ITEM_PROPERTIES_UPDATED', this.handleUpdatedProperties);
+        signalManager().on('EXPERIMENT_SELECTED', this.handleExperimentChanged);
+        signalManager().on('CLOSE_TRACEVIEWERTAB', this.handleTabClosed);
         return;
     }
 
-    handleExperimentChanged = (exp: Experiment) => {
-        const props = this.propertiesMap.get(exp?.UUID);
-        if (props) {
-            this.handleUpdatedProperties(props);
-        } else {
-            const emptyPayload = new ItemPropertiesSignalPayload({});
-            this.handleUpdatedProperties(emptyPayload);
-        }
+    handleExperimentChanged = (exp?: Experiment) => {
+        const payload = exp ? this.propertiesMap.get(exp.UUID) : new ItemPropertiesSignalPayload({});
```

Comment:
I am getting 'TypeError: Cannot read properties of undefined (reading 'getExperimentUUID')'

The payload could be undefined here - Either have to address it here or could handle it in handleUpdatedProperties

Reply:
Yea, there was a weird issue where when `exp` is `Experiment | undfined` you can't pass `string | undefined` into the getter.  So i changed the logic but made some the assumption we would always find something if exp was defined..

Anyway... I added another checker so if `this.propertiesMap.get(exp.UUID)` doesn't find anything we just put an empty `ItemPropertiesSignalPayload` which should fix that.